### ISSUE TITLE
[SlideFade] findDOMNode deprecation warning in StrictMode

### ISSUE THREAD

**Issue by selbekk** [chakra-ui/chakra-ui#1496]
**Describe the bug** 
findDOMNode warning when using StrictMode (v1 rc.0)

Reproduction: https://codesandbox.io/s/trusting-nightingale-kiij9?file=/src/index.tsx

To see the bug - toggle the animation and see the console.

This is because we don't use the `nodeRef` prop when using the `react-transition-group` (you can read more about the nodeRef prop in the [changelog](https://github.com/reactjs/react-transition-group/blob/1fd4a65ac45edd2aea3dec18eeb8b9c07c7eb93f/CHANGELOG.md#440-2020-05-05)) We can't really use it without changing the API either, so I can't really create a good PR to fix it without discussing the API implications with you.

One proposed API is changing the returned function value to include ref as well, changing the API from this

```tsx
<Fade in={isVisible}>
  {style => <Box style={style} />}
</Fade>
```

to this:

```tsx
<Fade in={isVisible}>
  {fadeProps => <Box {...fadeProps} />}
</Fade>
```

where `fadeProps` is an object containing the following:

```tsx
{
  style: style,
  ref: nodeRef,
}
```

**Comment by akiarostami** [comment]
I am having this issue with Drawer as well.

**Comment by with-heart** [comment]
imo this change to the api is fine, as long as we document it. It's not any more difficult to use than the previous.

**Comment by akiarostami** [comment]
Btw, where can I find the documentation about SlideFade?! I can't seem to find that anywhere.

**Comment by with-heart** [comment]
Hi @akiarostami! I created #1657 to get documentation added for the transition components.

**Comment by akiarostami** [comment]
Thank you @with-heart!

**Comment by FredyC** [comment]
I want to point out that the `Collapse` component is suffering from this warning as well, even though it is not part of the transitions package.

**Comment by aldrakos** [comment]
As a note, I am still seeing this error message the 1st time a `Drawer` opens (using the latest rc5). It does not seem to happen on the documentation page though (https://next.chakra-ui.com/docs/overlay/drawer)

**Comment by selbekk** [comment]
It doesn't happen in production, only during development.

**Comment by aldrakos** [comment]
> It doesn't happen in production, only during development.

That's good to know thanks :) Would still be helpful to not pop up in the final v1.0 release.

**Comment by ljosberinn** [comment]
Will most likely be fixed after merging #2200

**Comment by segunadebayo** [comment]
Considering our recent migration to `framer-motion`, this issue no longer exists.

**Comment by FredyC** [comment]
@segunadebayo Any ETA when it's going to be published? I am frankly a bit lost, I see RC4 in changelog while RC5 is out.

**Comment by ljosberinn** [comment]
Looks like the changelog hasn't been update for RC5, that's unfortunate. Here's an excerpt I manually posted on Discord at the time:

1. `rc.4` color mode issues should be resolved
2. `Checkbox` checked icon can be customized via `icon` prop
3. Node v10 compatibility restored
4. iOS 13 media query list issue fixed
5. `ColorModeScript` prop for color mode is now called `initialColorMode` just like in `theme.config`
6. docs are updated

1. and 4. had issues however, they have been fixed since.
